### ISSUE TITLE
Cloudflare Adapter V11 breaks AstroDB

### ISSUE THREAD

**Issue by MasterQuestMaster** [withastro/adapters#347]
### Astro Info

```block
Astro                    v4.13.1
Node                     v22.0.0
System                   Windows (x64)
Package Manager          npm
Output                   server
Adapter                  @astrojs/cloudflare
Integrations             astro:db
                         @astrojs/db/file-url
```

### Describe the Bug

While using AstroDB with the Cloudflare adapter, when starting the development server, after a new local database was created, the following error appears: 

```
21:07:11 [astro:db] New local database created.
21:07:12 [ERROR] [astro:db] Cannot find module 'astro:db' imported from '[path]/cloudflare-astro-test/db/seed.ts'
```
The development server is successfully started, but the "content.db" is empty.

At first I thought this had something to do with installing wrangler stuff, but even without that, the minimal example still has the error. So it's very likely the upgrade of the cloudflare adapter to V11 that breaks things.

### What's the expected result?

I expect that the seed.ts can be successfully executed and add data to the content.db.

### Link to Minimal Reproducible Example

https://github.com/MasterQuestMaster/cloudflare-astro-test

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by alexanderniebuhr** [comment]
Thanks for the report, I don't know where you got the information about updating the `tsconfig.json` from, but it's wrong.

```diff
{
  "name": "cloudflare-astro-test",
  "type": "module",
  "version": "0.0.1",
  "scripts": {
    "dev": "astro dev",
    "start": "astro dev",
    "build": "astro check && astro build",
    "preview": "astro build && wrangler pages dev",
    "astro": "astro",
    "deploy": "astro build && wrangler pages deploy",
    "cf-typegen": "wrangler types"
  },
  "dependencies": {
    "@astrojs/check": "^0.9.1",
    "@astrojs/cloudflare": "^11.0.4",
    "@astrojs/db": "^0.12.0",
    "astro": "^4.13.1",
    "typescript": "^5.5.4"
  },
  "devDependencies": {
-    "@cloudflare/workers-types": "^4.20240729.0",
    "wrangler": "^3.68.0"
  }
}
```

```diff
{
"extends": "astro/tsconfigs/strict",
"compilerOptions": {
-    "types": [
-        "@cloudflare/workers-types/2023-07-01"
-    ]
}
}
```

Remove the package and the `types` configuration from the example project and it starts working again.

**Comment by MasterQuestMaster** [comment]
@alexanderniebuhr The "updated" code of tsconfig.json is like that by default if you create a project using "npm create cloudflare" with the astro framework starter. The cloudflare installer CLI even has a dedicated section towards installing those worker types.

I have done as you advised and removed the worker types dependency and tsconfig update (and ran npm install afterwards to update). However, unlike what you suggested in your message, this did not fix the issue. I'm still getting the same error message. I tried removing package-lock and the node_modules folder and installing again, but same result.

I have updated the repo with the changes. Are you saying it works for you when you run it like that?
